Add release notes for 16.14 (#11858)
chef/chef: RELEASE_NOTES.md

In Chef Infra Client 16 we introduced Unified Mode allowing you to collapse the sometimes confusing compile and converge phases into a single unified phase. Unified mode makes it easier to write and troubleshoot failures in custom resources and for Chef Infra Client 18 we plan to make this the default execution phase for custom resources. We've backported the unified mode feature to the Chef Infra Client 14 and 15 systems and for Chef Infra Client 17 we will now begin warning if resources don't explicitly set this new mode. Enabling unified mode now lets you validate that resources will continue to function as expected in Chef Infra Client 18. To enable unified  mode in your resource add `unified_mode true` to the file.

## What's New in 16.14

## Bug Fixes

> - `bundle install` when installing gems from cookbook `metadata.rb` files. Thanks for this fix [@nvwls](https://github.com/nvwls)

Reviewer: Not sure of the context of this one, but

```suggestion
- `bundle install` now correctly installs gems from cookbook `metadata.rb` files. Thanks for this fix [@nvwls](https://github.com/nvwls)
```

- `knife bootstrap` on Windows now correctly uses `https://omnitruck.chef.io` to download packages.

### Chef InSpec 4.31

Chef InSpec has been updated from 4.31.1 to 4.38.9 with the following changes:

#### New Features

> - Added the new --reporter-include-source CLI option, which includes the source code of the controls in the output of the CLI reporter
> - Added ability to pass inputs to InSpec shell using input file and cli
> - Added a new mongodb_conf resource
> - Fixed the inspec shell to allow loading profiles that have their own dependent profiles
> - Updated the inspec init plugin command with the following changes
>   - The values of flags passed to the inspec init plugin command are now wrapped in double quotes instead of single quotes.
>   - Template files are now ERB files.
>   - The activator flag replaces the hook flag, which is now an alias.
> - Added support for zfs_pool and zfs_dataset resources on Linux
> - Improved port resource performance: adding more specific search while using ss command
> - The new inspec automate command replaces the inspec compliance command, which is now deprecated
> - Added the selinux resource which includes support for modules and booleans.
> - Added the pattern input option for DSL and metadata inputs.
> - Added the members_array property for group & groups resources.
> - Train now reads the username and port from the .ssh/config file and will use these values if present.

Reviewer: ```suggestion
- Added the new `--reporter-include-source` CLI option, which includes the source code of the controls in the output of the CLI reporter.
- Added ability to pass inputs to InSpec shell using input file and CLI.
- Added a new mongodb_conf resource.
- Fixed the inspec shell to allow loading profiles that have their own dependent profiles.
- Updated the inspec init plugin command with the following changes.
  - The values of flags passed to the inspec init plugin command are now wrapped in double quotes instead of single quotes.
  - Template files are now ERB files.
  - The activator flag replaces the hook flag, which is now an alias.
- Added support for zfs_pool and zfs_dataset resources on Linux.
- Improved port resource performance: adding more specific search while using ss command.
- The new inspec automate command replaces the inspec compliance command, which is now deprecated.
- Added the selinux resource which includes support for modules and booleans.
- Added the pattern input option for DSL and metadata inputs.
- Added the members_array property for group & groups resources.
- Train now reads the username and port from the .ssh/config file and will use these values if present.
```

#### Bug Fixes

> - Removed the default of 3600 seconds for --command-timeout CLI option
> - Fix SSH Timeout PTY allocation
> - Changed the Windows local pipe server connection to retry once on EPIPE
> - Fixed the postgres_session resource to raise an exception if there is an error in a connection or query
> - Fixed the mysql_session resource to raise an exception if there is an error in a connection or query
> - Removed support for Chef Compliance Server and Chef Automate 1 from the inspec automate command, as both products are EOL
> - inspec detect --no-color returns color-free output
> - file resource more_permissive_than matcher returns nil instead of throwing exception when file does not exist
> - The HTTP resource response body is now coerced into UTF-8.
> - Modified the windows_feature resource to indicate if a feature is enabled rather than just available.
> - Fixed an error when using profile dependencies and require_controls.
> - Fixed the windows_firewall_rule resource when it failed to validate more than one rule.
> - Switch to GNU timeout-based implementation of SSH timeouts
> - Fixed the group resource when a member does not exist.

Reviewer: ```suggestion
- Removed the default of 3600 seconds for `--command-timeout` CLI option.
- Fix SSH Timeout PTY allocation.
- Changed the Windows local pipe server connection to retry once on EPIPE.
- Fixed the postgres_session resource to raise an exception if there is an error in a connection or query.
- Fixed the mysql_session resource to raise an exception if there is an error in a connection or query.
- Removed support for Chef Compliance Server and Chef Automate 1 from the inspec automate command, as both products are EOL.
- `inspec detect --no-color` returns color-free output.
- file resource more_permissive_than matcher returns nil instead of throwing exception when file does not exist.
- The HTTP resource response body is now coerced into UTF-8.
- Modified the windows_feature resource to indicate if a feature is enabled rather than just available.
- Fixed an error when using profile dependencies and require_controls.
- Fixed the windows_firewall_rule resource when it failed to validate more than one rule.
- Switch to GNU timeout-based implementation of SSH timeouts.
- Fixed the group resource when a member does not exist.
```
